core/shared/src/main/scala/kantan/csv/HeaderDecoder.scala

```diff
@@ -36,18 +36,18 @@ object HeaderDecoder extends GeneratedHeaderDecoders {
   /** Summons an implicit instance of [[HeaderDecoder]] if one can be found, fails compilation otherwise. */
   def apply[A](implicit ev: HeaderDecoder[A]): HeaderDecoder[A] = macro imp.summon[HeaderDecoder[A]]
 
-  // Horrible code that relies on mutability and assumes that `map` contains the right values, indexed from 0 to length.
-  private[csv] def decoderFromMap(map: Map[String, Int], header: Seq[String]): DecodeResult[Array[Int]] = {
-    val mapping = Array.fill(map.size)(-1)
-
-    header.zipWithIndex.foreach {
-      case (name, i) =>
-        map.get(name).foreach(a => mapping(a) = i)
+  private def determineRowMappings(requiredHeader: Seq[String], csvHeader: Seq[String]): DecodeResult[Seq[Int]] = {
```

Comment:
Your problem here is that this should be package private, not private. See the modifier used for `decoderFromMap`.

Note that it might not need to be private anything - the method it replaces was because of how many unreasonable assumptions it made, so I wanted to be the only one to be able to call it, but this is perfectly safe, as far as I can tell.